packages/spanner/src/database.js

```diff
@@ -598,11 +633,18 @@ Database.prototype.runStream = function(query, options) {
     session: this.formattedName_
   });
 
+  var fields = {};
+
   if (reqOpts.params) {
-    var fields = {};
+    reqOpts.types = reqOpts.types || {};
 
     for (var prop in reqOpts.params) {
       var field = reqOpts.params[prop];
+
+      if (!reqOpts.types[prop]) {
```

Comment:
I think we need handling here if a user provides an invalid type, we should throw.